Airbnb.Application/Services/AuthService.cs

```diff
@@ -1,34 +1,48 @@
-using Airbnb.Application.DTOs;
+using System.Text.Json;
+using Airbnb.Application.DTOs.Authorization;
 using Airbnb.Domain.Entities;
 using Microsoft.AspNetCore.Identity;
-using Mapster;
 using Airbnb.Application.Interfaces;
 using Airbnb.Application.Interfaces.Services;
+using Airbnb.Application.Options;
+using Airbnb.Application.CreatedEvent;
+using Mapster;
 
 namespace Airbnb.Application.Services;
 
 public class AuthService : IAuthService
 {
     private readonly IUserManagerWrapper _userManagerWrapper;
     private readonly IJwtTokenService _jwtTokenService;
-    
-    public AuthService(IUserManagerWrapper userManagerWrapper, IJwtTokenService jwtTokenService)
+    private readonly IEventSender _eventSender;
+     
+    public AuthService(IUserManagerWrapper userManagerWrapper, IJwtTokenService jwtTokenService, 
+        IEventSender eventSender)
     {
         _userManagerWrapper = userManagerWrapper;
         _jwtTokenService = jwtTokenService;
-    }
+        _eventSender = eventSender;
+     }
 
     public async Task<IdentityResult> RegisterUserAsync(RegisterDto dto)
     {
         var user = dto.Adapt<ApplicationUser>();
         user.UserName = dto.Email;
         user.ExternalId = Guid.NewGuid().ToString();
+        user.CreatedAt = DateTime.UtcNow;
  
         var result = await _userManagerWrapper.CreateAsync(user, dto.Password);
 
         if (result.Succeeded)
         {
             await _userManagerWrapper.AddToRoleAsync(user, dto.Role);
+
+            var userEvent = user.Adapt<UserCreatedEvent>();
+            
+            var jsonMessage = userEvent;
+            var key = user.Id.ToString();
+            
+            await _eventSender.SendEvent(KafkaTopics.UserProfileUpdated, key, jsonMessage);
```

Comment:
Let's rework it:
1. event publisher accepts the generic TEvent and key
2. Inside, it decides based on the typeof(TEvent) to which topic it should be published (using your topic registry static class
3. Now you event publisher is generic and you don't pass topic name as parameter for publishing